test/mpi/pt2pt/pingping.c

```diff
@@ -41,75 +67,88 @@ int main(int argc, char *argv[])
          * change the error handler to errors return */
         MPI_Comm_set_errhandler(comm, MPI_ERRORS_RETURN);
 
-        for (count = 1; count < MAX_COUNT; count = count * 2) {
-
-            /* To shorten test time, only run the default version of datatype tests
-             * for comm world and run the minimum version for other communicators. */
-            if (comm != MPI_COMM_WORLD) {
-                MTestInitMinDatatypes();
+        for (i = 0; i < send_dtp->DTP_num_objs; i++) {
+            err = DTP_obj_create(send_dtp, i, 0, 1, count[0]);
+            if (err != DTP_SUCCESS) {
```

Comment:
I would prefer an "object-based" interface rather than the current "index-based" one. The object-based one is more straightforward to me as you "create an object from the pool". Moreover, you do not need to "translate" index into object at every user call.

The changes are:
- User passes the `index` at `obj_create`, and receives the datatype object (e.g., `send_objptr`).
- When accessing buffer, count information
```sendcount = send_dtp->DTP_obj_array[i].DTP_obj_count;```
becomes 
```sendcount = send_objptr->DTP_obj_count;```
- when free the object,
```DTP_obj_free(send_dtp, i);```
becomes
```DTP_obj_free(&send_objptr);```

@gcongiu and I have already discussed this change. @pavanbalaji : could you let us know your thought ?

Reply:
@minsii can this be left out for a future release? I am trying to keep changes to minimum since time left is not much